### ISSUE TITLE
Examples showing how to use auth modules for social logins

### ISSUE THREAD

**Issue by subhendukundu** [nhost/nhost-dart#1]
I think it would be beneficial if we add different examples for all the possible logins. Clear and easy to understand.

**Comment by shyndman** [comment]
There is [an example here](https://github.com/nhost/nhost-dart/blob/main/packages/nhost_flutter_auth/example/lib/oauth_providers_example.dart) for logging in via GitHub. The other providers should be similar. 

Does that help?

**Comment by subhendukundu** [comment]
It does definitely help. But I think it's just easier for developers, easy with sample code. Sometimes they can just stuck with small things. Plus I think we should just focus on "making product as easy as possible". So they can just focus on product. We can take care of the rest. Just a thought, you don't have to take this too seriously. 🙂

**Comment by shyndman** [comment]
No, it's a great suggestion, and I agree the process should be easier.

One of the tricky parts about the social logins is that you need to tweak the configuration for the native app, so it's not quite as copy-pasteable as other examples, even when all the Dart code is right in front of you. And then there's also the backend config, and having to set that up properly.

Maybe the ideal would be to add some capability to `nhost_flutter_auth`, or a new package (one that includes the necessary plugins). An API where you could say `nhost.auth.loginWithGitHub()`.

Maybe the package could even check your configuration (native app, and backend). Complain if it's wrong?

**Comment by subhendukundu** [comment]
So We can keep 2 packages. One is the core. Few devs might not want to add these but we will still add these dependencies, which is not good I think.
The other package/packages, which will have all the native, and devs just plug it in and use it. Might be okay adding easy config in native if really needed to.

@elitan and I keen to make developer experiences as easy we can. Just like no-code but legos.

**Comment by shyndman** [comment]
> @elitan and I keen to make developer experiences as easy we can. Just like no-code but legos.

Makes sense. I think we can get something like that going.

**Comment by shyndman** [comment]
Sorry, to be clear, that seems like a good goal, and is certainly possible. Whether to pursue is not my call.

**Comment by subhendukundu** [comment]
Well I had been playing around with nhost-dart for a while now. Few observations.

1. Using app_links for the login, very smart move but I haven't able to make it work with.
2. Docs we need to work on.

So this is what I did.

1. I have used `flutter_web_auth`.
Setting the callback urls to `'io.higgle.dakuapp://login-callback`
```
Future googleSignIn() async {
    try {
      final result = await FlutterWebAuth.authenticate(
        url: nhostGoogleLoginUrl,
        callbackUrlScheme: 'io.higgle.dakuapp://login-callback',
      );
      nhostClient.auth.completeOAuthProviderLogin(Uri.parse(result));
      return true;
    } catch (e) {
      print(e);
      return false;
    }
  }
  ```
And this works. But what about the web support? https://daku.app has web support. Any idea?
